### PR TITLE
RELEASE_NOTES.rst: Add 0.11.0 release notes

### PR DESCRIPTION
**Pull request by meetmangukiya**
@meetmangukiya summons @Mixih - the release lead for a review

and these are the stats, go figure.
```sh
meet@phoenix ~/Desktop/ghubby/coala/coala-bears [meet/relnotes/0.11 *]
_ % git log --oneline 1096b293af15059ce363c0569d5927de2cefd9c2..master | wc                                                                                       !3862
     25     131    1263

meet@phoenix ~/Desktop/ghubby/coala/coala-bears [meet/relnotes/0.11 *]
_ % git log --oneline rel/0.10.1..master | wc                                                                                                                     !3863
    101     529    4884
```
I can't believe I've written release notes for 76 commits :P

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+
+	**CSS**
+	- `StyleLintBear` - Lints css files using stylelint
+
+	**Generic**
+	- `LicenseCheckBear` - Checks given files for license searching start or end of files for lines belonging to various licenses.
+	- `DuplicateFileBear` - Checks for duplicate files.
+
+	**Elm**
+	- `ElmLinkBear` - Formats source code according to given set of rules.
```
Comment (yash-nisar) [inline RELEASE_NOTES.rst]: `ElmLinkBear` -> `ElmLintBear`

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -1,3 +1,39 @@
+coala-bears 0.10.0
+==================
+
+<!-- upto 1096b293af15059ce363c0569d5927de2cefd9c2 -->
+**Bug Fixes** 
+
+- `InvalidLinkBear` now extracts links from markdown correctly. [Issue 1338](https://github.com/coala/coala-bears/issues/1338)
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: \` - > \`\` (rst syntax)

Comment (Mixih) [inline RELEASE_NOTES.rst]: put the issue mention on a new line please, it looks visually cleaner int he code.

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+- `InvalidLinkBear` now extracts all the links in a line. Earlier it used to extract only the first match.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see backtick note above

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -1,3 +1,39 @@
...
+- `JSONFormatBear` notifies if run on empty files.
+- `JSONFormatBear` tells the lines and cols at which the error occured instead of showing it in the error message.
+- `GitCommitBear` can now enforce issue reference in commits.
+- `GitCommitBear` now recognizes `user/repo#12` type issue references.
+- `LineCountBear` can now exclude blank lines in counting the number of lines in a file. The new setting name is `exclude_blank_lines` and it is disabled by default i.e. `False`.
+- `FilenameBear` can now recognize kebabcase naming convention.
+- `FilenameBear` can predict naming conventions for wide-spread languages like java, javascript and python. Set the `file_naming_convention` to `auto` to use this feature.
+- `ShellCheckBear` can ignore specific rules by providing those in `ignore_actions` setting.
+
+**Bears with changed default settings**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: any deprecations?

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -1,3 +1,39 @@
...
+	**Elm**
+	- `ElmLintBear` - Formats source code according to given set of rules.
+
+**Enhancements**
+- `JSONFormatBear` notifies if run on empty files.
+- `JSONFormatBear` tells the lines and cols at which the error occured instead of showing it in the error message.
+- `GitCommitBear` can now enforce issue reference in commits.
+- `GitCommitBear` now recognizes `user/repo#12` type issue references.
+- `LineCountBear` can now exclude blank lines in counting the number of lines in a file. The new setting name is `exclude_blank_lines` and it is disabled by default i.e. `False`.
+- `FilenameBear` can now recognize kebabcase naming convention.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: alphabetical please

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -1,3 +1,39 @@
...
+- `FilenameBear` can predict naming conventions for wide-spread languages like java, javascript and python. Set the `file_naming_convention` to `auto` to use this feature.
+- `ShellCheckBear` can ignore specific rules by providing those in `ignore_actions` setting.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: in `ignore_actions` -> in the ``ignore_actions``

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -1,3 +1,39 @@
...
+	**Generic**
+	- `LicenseCheckBear` - Checks given files for license searching start or end of files for lines belonging to various licenses.
+	- `DuplicateFileBear` - Checks for duplicate files.
+
+	**Elm**
+	- `ElmLintBear` - Formats source code according to given set of rules.
+
+**Enhancements**
+- `JSONFormatBear` notifies if run on empty files.
+- `JSONFormatBear` tells the lines and cols at which the error occured instead of showing it in the error message.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: please linebreak at 80 if you can

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -1,3 +1,39 @@
+coala-bears 0.10.0
+==================
+
+<!-- upto 1096b293af15059ce363c0569d5927de2cefd9c2 -->
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: note for when you get this done: contributor count, commit count ;)

File: RELEASE_NOTES.rst [hunk 9]
```diff
@@ -1,3 +1,39 @@
...
+**Bug Fixes** 
+
+- `InvalidLinkBear` now extracts links from markdown correctly. [Issue 1338](https://github.com/coala/coala-bears/issues/1338)
+- `InvalidLinkBear` now extracts all the links in a line. Earlier it used to extract only the first match.
+- `PyCodeStyleBear` now uses user settings along with default settings. Earlier it used to ignore default settings if user changed one of them.
+
+**New Bears**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: follow the old format please (c.f. line 57)

File: RELEASE_NOTES.rst [hunk 10]
```diff
@@ -1,3 +1,39 @@
...
+- `JSONFormatBear` tells the lines and cols at which the error occured instead of showing it in the error message.
+- `GitCommitBear` can now enforce issue reference in commits.
+- `GitCommitBear` now recognizes `user/repo#12` type issue references.
+- `LineCountBear` can now exclude blank lines in counting the number of lines in a file. The new setting name is `exclude_blank_lines` and it is disabled by default i.e. `False`.
+- `FilenameBear` can now recognize kebabcase naming convention.
+- `FilenameBear` can predict naming conventions for wide-spread languages like java, javascript and python. Set the `file_naming_convention` to `auto` to use this feature.
+- `ShellCheckBear` can ignore specific rules by providing those in `ignore_actions` setting.
+
+**Bears with changed default settings**
+- `YAMLLintBear` - `document_start` setting is changed to `None`, not forbiding the usage of document start marker(---) by default. [Issue 1417](https://github.com/coala/coala-bears/issues/1417)
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: linebreak at 80 please

File: RELEASE_NOTES.rst [hunk 11]
```diff
@@ -1,3 +1,39 @@
+coala-bears 0.10.0
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: 0.111.0 ;) I forgot to merge the previous relnotes

File: RELEASE_NOTES.rst (hunk 2 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: damn, forgot

File: RELEASE_NOTES.rst (hunk 4 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: I don't think so because I haven't added it :/

Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: there is a deprecation

File: RELEASE_NOTES.rst [hunk 12]
```diff
@@ -1,3 +1,64 @@
+coala-bears 0.11.0
+==================
+
+This release has been brought to you by 30 contributor contributing over 101
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: s/contributor/contributors

File: RELEASE_NOTES.rst [hunk 13]
```diff
@@ -6,14 +67,14 @@ Bugfixes:
...
- 
+
 coala-bears 0.10.0
 ==================
 
 This release, we had 97 commits contributed by 42 contributors.
 
 The coala website has been updated to a new, more modern design. You can now
-find a quick overview of bears and their settings at 
+find a quick overview of bears and their settings at
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: Split these to a different commit please

File: RELEASE_NOTES.rst [hunk 14]
```diff
@@ -1,3 +1,64 @@
...
+- ``InvalidLinkBear`` now extracts all the links in a line. Earlier it used to
+  extract only the first match.
+- ``PyCodeStyleBear`` now uses user settings along with default settings.
+  Earlier it used to ignore default settings if user changed one of them.
+- ``PyImpostSortBear`` now recongnizes known third party imports correctly.
+  [Issue 1510](https://github.com/coala/coala-bears/issues/1510)
+
+**New Bears**
+
+	**CSS**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: no indent or bolds here please or any other similar places. (please match previous format)

File: RELEASE_NOTES.rst [hunk 15]
```diff
@@ -1,3 +1,64 @@
...
+	- ``StyleLintBear`` - Lints css files using stylelint
+
+	**Elm**
+	- ``ElmLintBear`` - Formats source code according to given set of rules.
+
+	**Generic**
+	- ``LicenseCheckBear`` - Checks given files for license searching start or
+      end of files for lines belonging to various licenses.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: indent is off here

File: RELEASE_NOTES.rst [hunk 16]
```diff
@@ -1,3 +1,64 @@
...
+	- ``DuplicateFileBear`` - Checks for duplicate files.
+
+**Enhancements**
+- ``FilenameBear`` can now recognize kebabcase naming convention.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: s/kebabcase/kebab-case

File: RELEASE_NOTES.rst [hunk 17]
```diff
@@ -1,3 +1,64 @@
...
+- ``FilenameBear`` can predict naming conventions for wide-spread languages like
+  java, javascript and python. Set the ``file_naming_convention`` to ``auto`` to
+  use this feature.
+- ``GitCommitBear`` can now enforce issue reference in commits.
+- ``GitCommitBear`` now recognizes ``user/repo#12`` type issue references.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: issue link or more clarification please. I don't quite understand this.

File: RELEASE_NOTES.rst [hunk 18]
```diff
@@ -1,3 +1,64 @@
...
+- ``JSONFormatBear`` notifies if run on empty files.
+- ``JSONFormatBear`` tells the lines and cols at which the error occured instead
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: s/cols/columns (generally please don't abbreviate in relnotes, especially non-standard ones)

File: RELEASE_NOTES.rst [hunk 19]
```diff
@@ -1,3 +1,64 @@
...
+  of showing it in the error message.
+- ``LineCountBear`` can now exclude blank lines in counting the number of lines
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: maybe move this to a "New Settings" section

File: RELEASE_NOTES.rst [hunk 20]
```diff
@@ -1,3 +1,64 @@
...
+  in a file. The new setting name is ``exclude_blank_lines`` and it is disabled
+  by default i.e. ``False``.
+- ``PyUnusedCodeBear`` can remove unused variables, see setting
+  ``remove_unused_variable``.
+- ``ShellCheckBear`` can ignore specific rules by providing those in
+  the ``ignore_actions`` setting.
+- ``YAMLLintBear`` can now be configured to check for maximum characters in a
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see `LineCountBear` comment

File: RELEASE_NOTES.rst [hunk 21]
```diff
@@ -1,3 +1,64 @@
+coala-bears 0.11.0
+==================
+
+This release has been brought to you by 30 contributor contributing over 101
+commits.
+
+**Bug Fixes**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: please move this section to the bottom to be consistent with the previous relnotes

File: RELEASE_NOTES.rst [hunk 22]
```diff
@@ -1,3 +1,64 @@
...
+	- ``ElmLintBear`` - Formats source code according to given set of rules.
+
+	**Generic**
+	- ``LicenseCheckBear`` - Checks given files for license searching start or
+      end of files for lines belonging to various licenses.
+	- ``DuplicateFileBear`` - Checks for duplicate files.
+
+**Enhancements**
+- ``FilenameBear`` can now recognize kebabcase naming convention.
+- ``FilenameBear`` can predict naming conventions for wide-spread languages like
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see LineCountBear comment

File: RELEASE_NOTES.rst [hunk 23]
```diff
@@ -1,3 +1,64 @@
...
+  use this feature.
+- ``GitCommitBear`` can now enforce issue reference in commits.
+- ``GitCommitBear`` now recognizes ``user/repo#12`` type issue references.
+- ``JSONFormatBear`` notifies if run on empty files.
+- ``JSONFormatBear`` tells the lines and cols at which the error occured instead
+  of showing it in the error message.
+- ``LineCountBear`` can now exclude blank lines in counting the number of lines
+  in a file. The new setting name is ``exclude_blank_lines`` and it is disabled
+  by default i.e. ``False``.
+- ``PyUnusedCodeBear`` can remove unused variables, see setting
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: new setting?

File: RELEASE_NOTES.rst [hunk 24]
```diff
@@ -1,3 +1,64 @@
...
+  ``remove_unused_variable``.
+- ``ShellCheckBear`` can ignore specific rules by providing those in
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: new setting or old setting new features?

File: RELEASE_NOTES.rst [hunk 25]
```diff
@@ -1,3 +1,64 @@
...
+  the ``ignore_actions`` setting.
+- ``YAMLLintBear`` can now be configured to check for maximum characters in a
+  line, by default it is set to 80.
+
+**Bears with changed default settings**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: perhaps change to `Default Setting Modifications`?

File: RELEASE_NOTES.rst [hunk 26]
```diff
@@ -1,3 +1,64 @@
...
+- ``LanguageToolBear`` - ``language`` setting is changed to
+  ``natural_language``, please update your settings, it'll be deprecated
+  completely in next release.
+- ``MarkdownBear`` - ``max_line_length`` setting is changed to ``None``, now it
+  is not mandatory to wrap at 80.
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: `now it is not mandatory to wrap at 80` -> `instead of 80`

File: RELEASE_NOTES.rst [hunk 27]
```diff
@@ -1,3 +1,64 @@
...
+**New Bears**
+
+	**CSS**
+	- ``StyleLintBear`` - Lints css files using stylelint
+
+	**Elm**
+	- ``ElmLintBear`` - Formats source code according to given set of rules.
+
+	**Generic**
+	- ``LicenseCheckBear`` - Checks given files for license searching start or
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: period after `license`, rewrite the beginning of the second line to work as a new sentence please

File: RELEASE_NOTES.rst [hunk 28]
```diff
@@ -1,3 +1,64 @@
...
+  Earlier it used to ignore default settings if user changed one of them.
+- ``PyImpostSortBear`` now recongnizes known third party imports correctly.
+  [Issue 1510](https://github.com/coala/coala-bears/issues/1510)
+
+**New Bears**
+
+	**CSS**
+	- ``StyleLintBear`` - Lints css files using stylelint
+
+	**Elm**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see above

File: RELEASE_NOTES.rst [hunk 29]
```diff
@@ -1,3 +1,64 @@
...
+	- ``ElmLintBear`` - Formats source code according to given set of rules.
+
+	**Generic**
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see above

File: RELEASE_NOTES.rst (hunk 23 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: yeah

File: RELEASE_NOTES.rst (hunk 24 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: new setting old feature

File: RELEASE_NOTES.rst [hunk 30]
```diff
@@ -1,3 +1,64 @@
...
+
+**Bug Fixes**
+
+- ``InvalidLinkBear`` now extracts links from markdown correctly.
+  [Issue 1338](https://github.com/coala/coala-bears/issues/1338)
+- ``InvalidLinkBear`` now extracts all the links in a line. Earlier it used to
+  extract only the first match.
+- ``PyCodeStyleBear`` now uses user settings along with default settings.
+  Earlier it used to ignore default settings if user changed one of them.
+- ``PyImpostSortBear`` now recongnizes known third party imports correctly.
```
Comment (NiklasMM) [inline RELEASE_NOTES.rst]: `PyImpostSortBear` -> `PyImportSortBear`

Consider making it more clear: "now correctly supports option 'know_third_party_imports'"

File: RELEASE_NOTES.rst (hunk 17 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: There is no relevant issue for this one, it was created in a PR of a different issue, https://github.com/coala/coala-bears/issues/1331

File: RELEASE_NOTES.rst (hunk 20 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: strange, I moved this, still the comment stays

File: RELEASE_NOTES.rst (hunk 24 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: same, I have moved this as well, still the comment stays :/

File: RELEASE_NOTES.rst [hunk 31]
```diff
@@ -1,3 +1,68 @@
...
+
+**Bug Fixes**
+
+- ``InvalidLinkBear`` now extracts links from markdown correctly.
+  [Issue 1338](https://github.com/coala/coala-bears/issues/1338)
+- ``InvalidLinkBear`` now extracts all the links in a line. Earlier it used to
+  extract only the first match.
+- ``PyCodeStyleBear`` now uses user settings along with default settings.
+  Earlier it used to ignore default settings if user changed one of them.
+- ``PyImportSortBear`` now correctly suppors option ``known_third_party_imports``.
```
Comment (NiklasMM) [inline RELEASE_NOTES.rst]: `suppors` -> `supports` ;-)

Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: too many typos, damn! #weNeedABearForTypos

**Comment by Mixih** [comment]
go go go

File: RELEASE_NOTES.rst [hunk 32]
```diff
@@ -1,3 +1,68 @@
+coala-bears 0.11.0
+==================
+
+This release has been brought to you by 30 contributors contributing over 101
+commits.
+
+**New Bears**
+
+CSS
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: newline after each heading please, otherwise rst complains

File: RELEASE_NOTES.rst [hunk 33]
```diff
@@ -1,3 +1,68 @@
...
+- ``StyleLintBear`` - Lints css files using stylelint
+
+Elm
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see previous

File: RELEASE_NOTES.rst [hunk 34]
```diff
@@ -1,3 +1,68 @@
...
+- ``ElmLintBear`` - Formats source code according to given set of rules.
+
+Generic
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see previous

File: RELEASE_NOTES.rst [hunk 35]
```diff
@@ -1,3 +1,68 @@
...
+- ``LicenseCheckBear`` - Checks given files for license. Searches start or
+  end of files for lines belonging to various licenses.
+- ``DuplicateFileBear`` - Checks for duplicate files.
+
+Enhancements
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: see previous

File: RELEASE_NOTES.rst [hunk 36]
```diff
@@ -1,3 +1,68 @@
...
+  line, by default it is set to 80.
+
+**Default Setting Modifications**
+
+- ``LanguageToolBear`` - ``language`` setting is changed to
+  ``natural_language``, please update your settings, it'll be deprecated
+  completely in next release.
+- ``MarkdownBear`` - ``max_line_length`` setting is changed to ``None`` instead
+  of 80.
+  [Issue 1581](https://github.com/coala/coala-bears/issues/1581)
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: link syntax is incorrect. please use `text <link>`_

File: RELEASE_NOTES.rst [hunk 37]
```diff
@@ -1,3 +1,68 @@
...
+- ``YAMLLintBear`` - ``document_start`` setting is changed to ``None``, not
+  forbiding the usage of document start marker(---) by default.
+  [Issue 1417](https://github.com/coala/coala-bears/issues/1417)
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: incorrect link syntax

File: RELEASE_NOTES.rst [hunk 38]
```diff
@@ -1,3 +1,68 @@
...
+
+**Bug Fixes**
+
+- ``InvalidLinkBear`` now extracts links from markdown correctly.
+  [Issue 1338](https://github.com/coala/coala-bears/issues/1338)
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: incorrect link syntax

File: RELEASE_NOTES.rst [hunk 39]
```diff
@@ -1,3 +1,68 @@
...
+- ``InvalidLinkBear`` now extracts all the links in a line. Earlier it used to
+  extract only the first match.
+- ``PyCodeStyleBear`` now uses user settings along with default settings.
+  Earlier it used to ignore default settings if user changed one of them.
+- ``PyImportSortBear`` now correctly supports option ``known_third_party_imports``.
+  [Issue 1510](https://github.com/coala/coala-bears/issues/1510)
```
Comment (Mixih) [inline RELEASE_NOTES.rst]: incorrect link syntax

File: RELEASE_NOTES.rst (hunk 36 above)
Comment (meetmangukiya) [inline RELEASE_NOTES.rst]: hmm. I remember you saying all markdown syntax works in rST

File: RELEASE_NOTES.rst [hunk 40]
```diff
@@ -1,3 +1,75 @@
...
+- ``InvalidLinkBear`` now extracts links from markdown correctly.
+  `Issue 1338 <https://github.com/coala/coala-bears/issues/1338>`_
+- ``InvalidLinkBear`` now extracts all the links in a line. Earlier it used to
+  extract only the first match.
+- ``PyCodeStyleBear`` now uses user settings along with default settings.
+  Earlier it used to ignore default settings if user changed one of them.
+- ``PyImportSortBear`` now correctly supports option ``known_third_party_imports``.
+  `Issue 1510 <https://github.com/coala/coala-bears/issues/1510>`_
+- ``KeywordBear`` doesn't report false results when unsetting it. Previously,
+  it used to match every alphabet instead of matching nothing.
```
Comment (jayvdb) [inline RELEASE_NOTES.rst]: every alphabet -> any text

File: RELEASE_NOTES.rst (hunk 13 above)
Comment (jayvdb) [inline RELEASE_NOTES.rst]: If this is being merged to master, a proper fix is to prevent it occurring again by setting up a bear.

**Comment by Mixih** [comment]
ack eee9d6e b924900

**Comment by Mixih** [comment]
oh wait, please change the base @meetmangukiya

**Comment by Mixih** [comment]
ack 886aba6 5cb7a70 9eddf44 1588a33 7a15c93

**Comment by Mixih** [comment]
@rultor merge
